Automattic/wp-calypso
# Account Recovery: redux state plumbing for <ResetPasswordSmsForm>

## Summary
This PR connects the underlying redux state to the `<ResetPasswordSmsForm>` so that it is functional. There are several things included to make it work:

1. Add necessary reducers, selectors and actions
1. Connect the underlying redux state to `<ResetPasswordSmsForm />` component
1. Extract the common error message as `<AccountRecoveryErrorMessage />` component

## Test Plan

1. Open an incognito window and make sure you are logged out.
1. Visit http://calypso.localhost:3000/account-recovery/reset-password/sms-form
1. Fill in the user info by running `dispatch( { type: actionTypes.ACCOUNT_RECOVERY_RESET_UPDATE_USER_DATA, userData: { user: 'southptestcall2' } } );` from the console. You can choose whichever test account as long as it has bound a 2FA phone number from http://calypso.localhost:3000/me/security/two-step
1. Fill in the reset method by running `dispatch( { type: actionTypes.ACCOUNT_RECOVERY_RESET_PICK_METHOD, method: 'primary_sms' } );` from the console.
1. Make a real request for the validation code via `dispatch( { type: actionTypes.ACCOUNT_RECOVERY_RESET_REQUEST, request: { user: 'southptestcall2', method: 'primary_sms' } } );`. After a few seconds, you should receive a validation code from a SMS message.
1. Fill in the received code and hit "continue". Nothing will happen from the UI side, but you can find a successful POST request to `/account-recovery/validate` endpoint from the network inspector.
1. Randomly fill in some gibberish code, and you should see an error message like this:
![image](https://cloud.githubusercontent.com/assets/1842898/24743144/bb89eff6-1adf-11e7-8c59-1396fbe2980d.png)
